server/controllers/login.js

```diff
@@ -0,0 +1,48 @@
+const bcrypt = require("bcrypt");
+const { sign } = require("jsonwebtoken");
+const getPassword = require("../database/queries/getPassword");
+require("dotenv").config();
+const { SECRET } = process.env;
+const createToken = (email, secret) => {
+  return sign({ email }, secret);
+};
+
+exports.login = (req, res) => {
+  const { email, password } = req.body;
+  if (!email || !password) {
+    return res
+      .status(200)
+      .json({ message: 'all fields are required', status: '400' });
+  }
+  getPassword(email, (err, result) => {
+    if (err) {
+      console.log(err);
+    } else {
+      if (result) {
+        const hash = result.password;
+        bcrypt
+          .compare(password, hash)
+          .then((result) => {
+            if (result) {
+              const token = createToken(email, SECRET);
+              res
+                .cookie('token', token, { maxAge: 900000, httpOnly: true })
+                .json({ status: 'sucess', token });
+            } else {
+              res
+                .status(200)
+                .json({
+                  message: 'incorrect email or password',
+                  status: '403'
+                });
+            }
+          })
+          .catch((err) => console.log(err));
+      } else {
+        res
+          .status(200)
+          .json({ message: 'incorrect email or password', status: '403' });
+      }
+    }
+  });
```

Comment:
this entire callback is difficult to read, too many nested ifs.. First, see my comment in the query below on using promises, then here's how it could look here:
```js
getPassword(email)
  .then(result => {
    if (!result || result.rows.length === 0) {
      throw { status: 403, message: 'incorrect email or password' }
    }
    const hash = result[0].password;
    return bcrypt.compare(password, hash);
  })
  .then(bcryptRes => {
    if (!res) {
      throw { status: 403, message: 'incorrect email or password' }
    }
    const token = createToken(email, SECRET);
    res
      .cookie('token', token, { maxAge: 900000, httpOnly: true })
      .status(200);
  })
  .catch(err => {
    if (err.status === 403) {
      res.status(403).json({ message: err.message });
    } else {
      console.log("Error occured in getPassword(): " + err);
      res.status(500).json({ message: "Unexpected error occured"});
    }
  })
```
the `catch` part could actually be extracted into a middleware and reused accross all of your controllers as a uniform error handler, however let's not get into that now. See if you can understand the code and let me know if you have any questions!